dask_sql/physical/rel/logical/limit.py

```diff
@@ -25,82 +22,12 @@ def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContai
         df = dc.df
         cc = dc.column_container
 
-        offset = rel.getOffset()
-        if offset:
-            offset = RexConverter.convert(offset, df, context=context)
-
-        end = rel.getFetch()
-        if end:
-            end = RexConverter.convert(end, df, context=context)
-
-            if offset:
-                end += offset
+        limit = RexConverter.convert(rel, rel.limit().getLimitN(), df, context=context)
 
-        df = self._apply_offset(df, offset, end)
+        # If an offset was present it would have already been processed at this point.
+        # Therefore it is always safe to start at 0 when applying the limit
+        df = df.head(limit, npartitions=-1, compute=False)
```

Comment:
My personal opinion here is to still check if the first partition has enough elements and if not call head with `npartitions=-1`.

@charlesbluca Do you think this is worth a broader issue/discussion to see how this can be optimized? 
For example in cases like this
```python
SELECT * from really_large_dataset LIMIT 100
```
is going to read every single partition if `npartitions=-1` before returning 100 rows which isn't ideal.